src/naf-dialog-adapter.js

```diff
@@ -143,26 +192,159 @@ export default class DialogAdapter {
+      // Send transport has been create but for whatever reason is stuck in new state and doesn't gather candidates.
+      if (
+        force ||
+        this._sendTransport.connectionState === "failed" ||
+        (!isFirefox && this._sendTransport.connectionState === "disconnected") ||
+        (this._sendTransport.connectionState === "new" && this._lastSendConnectionState === "failed")
+      ) {
+        this.emitRTCEvent("log", "RTC", () => `Restarting send transport ICE`);
+        if (this._protoo.connected) {
+          if (isFirefox) {
+            this.reconnect();
+          } else if (!this._sendTransport.closed) {
+            const { host, port, turn } = await window.APP.hubChannel.getHost();
+            const iceServers = this.getIceServers(host, port, turn);
+            await this._sendTransport.updateIceServers({ iceServers });
+            const iceParameters = await this._protoo.request("restartIce", { transportId: this._sendTransport.id });
+            await this._sendTransport.restartIce({ iceParameters });
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks the Send Transport ICE status and restarts it in case is in failed or disconnected states.
+   * This is called by the Send Transport "connectionstatechange" event listener.
+   * @param {boolean} connectionState The transport connnection state (ICE connection state)
+   */
+  checkSendIceStatus(connectionState) {
+    // If the ICE connection state failed we force an ICE negotiation
+    if (connectionState === "failed" || connectionState === "disconnected") {
+      this.restartSendICE(false);
+    } else if (connectionState === "connected") {
+      this._iceReconnectSendTaskId && clearTimeout(this._iceReconnectSendTaskId);
+      this._iceReconnectSendTaskId = null;
+    }
+
+    this._lastSendConnectionState = connectionState;
+  }
+
+  /**
+   * Starts a watchdog to monitorize the state of the Send Transport ICE connection state.
+   * @param {boolean} force Forces the execution of the reconnect.
+   */
+  startSendIceConnectionStateWd(force = false) {
+    if (force) {
+      this.reconnect();
+    } else {
+      if (!this._sendTaskId) {
+        this._sendTaskId = setInterval(() => {
+          this.restartSendICE(false);
+        }, ICE_RECONNECT_INTERVAL);
+      }
     }
   }
 
+  /**
+   * Stops the Send Transport ICE connection state watchdog.
+   */
+  stopSendIceConnectionStateWd() {
+    this._sendTaskId && clearInterval(this._sendTaskId);
+    this._sendTaskId = null;
+  }
+
   /**
    * Restart ICE in the underlying receive peerconnection.
+   * @param {boolean} force Forces the execution of the reconnect.
    */
-  async restartRecvICE() {
+  async restartRecvICE(force = true) {
+    // Renegotiate ICE in case ICE state is failed, disconnected. The last condition cover the cases where the
+    // Send transport has been create but for whatever reason is stuck in new state and doesn't gather candidates.
```

Comment:
```suggestion
    // send transport has been create but for whatever reason is stuck in new state and doesn't gather candidates.
```
semantic change, let's me know it's in the middle of the sentence, up to you to change.

Reply:
Updated